src/MacroTools/Spells/AddAbilityonCast.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System.Collections.Generic;
+using MacroTools.Buffs;
+using MacroTools.Extensions;
+using MacroTools.SpellSystem;
+using WCSharp.Buffs;
+using WCSharp.Shared.Data;
+using static War3Api.Common;
+
+namespace MacroTools.Spells
+{
+  /// <summary>
+  /// add specific abilities to the caster for a limited duration.
+  /// </summary>
+  public sealed class AddAbilityonCast : Spell
+  {
+    public float DurationBase { get; init; }
```

Comment:
We use `LeveledAbilityProperty` for levelled fields like this now.
```suggestion
    public LeveledAbilityProperty<float> Duration { get; init; } = new();
```
Then you can delete the `DurationLevel` one.

Reply:
This isn't resolved as the `DurationLevel` property still exists.